src/Microsoft.OData.Core/ODataMessageWriterSettings.cs

```diff
@@ -182,6 +183,11 @@ public ODataUri ODataUri
         /// </summary>
         internal bool ThrowOnUndeclaredPropertyForNonOpenType { get; private set; }
 
+        /// <summary>
+        /// Returns whether ThrowOnTopLevelNullProperty validation setting is enabled.
+        /// </summary>
+        internal bool ThrowOnTopLevelNullProperty { get; private set; }

```

Comment:
This new field should also be included in the CopyFrom method of this class.

Reply:
Done. Also added tests for it.